javatests/arcs/core/analysis/testdata/ok_field_claim_propagates_type_variables.arcs

```diff
@@ -0,0 +1,27 @@
+// #Ingress: P1
+// #OK
+particle P1
+  foo: writes Foo {a: Text, b: Number}
+  claim foo.a is trusted
+  claim foo.b is untrusted
+particle P2
+  foo: reads ~a with {a: Text} // Constraints
+  bar: writes ~b
+particle P3
+  baz: reads Bar {b: Text}
+  foo: writes [~a]
+  check baz.b is trusted
+particle P4
+  foo: reads [Foo {b: Number}]
+  check foo.b is trusted
+recipe R
+  P1
+    foo: writes h1
+  P2
+    foo: reads h1
+    bar: writes h2
+  P3
+    baz: reads h2
+    foo: writes h3
+  P4
```

Comment:
maybe:

```
particle P1
  foo: writes Foo {a: Text, b: Number}
  claim foo.a is trusted
  claim foo.b is untrusted
particle P2
  foo: reads ~a with {a: Text}
  bar: writes Foo {a: Text, b: Number}
particle P3
  foo: reads Foo {a: Text, b: Number}
  check foo.b is trusted
recipe R
  P1
    foo: writes h1
  P2
    foo: reads h1
    bar: writes h2
  P3
    foo: reads h2
```

It is interesting, because if you connect P1 to P3 it would fail, but P2 is routing the data from Foo.a into Foo.{a, b}, which makes Foo.b trusted, even though it was untrusted in the P1.

Reply:
I wanted to be able to test both the read and write connection with a type variable. Is the following snippet reasonable?
```
 ...
particle P2
  foo: reads ~a with {a: Text}  // Constraints
  bar: writes Bar {b: Text}
particle PassThrough
  input: reads ~a
  output: writes ~a
particle P3
  baz: reads Bar {b: Text}
  foo: writes [Foo {b: Number}]
  check baz.b is trusted
 ...
```

Reply:
Yes, that looks ok.